bigquery/snippets/managedwriter/bigquerystorage_append_rows_pending.go

```diff
@@ -0,0 +1,205 @@
+// to a pending stream, and then committing it to a table.
+func appendToPendingStream(w io.Writer, projectID, datasetID, tableID string) error {
+	// projectID := "myproject"
+	// datasetID := "mydataset"
+	// tableID := "mytable"
+
+	ctx := context.Background()
+	client, err := managedwriter.NewClient(ctx, projectID)
+	if err != nil {
+		return fmt.Errorf("managedwriter.NewClient: %v", err)
+	}
+
+	// Create a new pending stream.  We'll use the stream name to construct a writer.
+	pendingStream, err := client.CreateWriteStream(ctx, &storagepb.CreateWriteStreamRequest{
+		Parent: fmt.Sprintf("projects/%s/datasets/%s/tables/%s", projectID, datasetID, tableID),
+		WriteStream: &storagepb.WriteStream{
+			Type: storagepb.WriteStream_PENDING,
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("CreateWriteStream: %v", err)
+	}
+
+	// We need to communicate the descriptor of the protocol buffer message we're using, which
+	// is analagous to the "schema" for the message.  This example uses a self-contained message, but
+	// we'll call adapt.NormalizeDescriptor which will correct any descriptors that aren't
+	// self-contained.
+	m := &exampleproto.SampleData{}
+	descriptorProto, err := adapt.NormalizeDescriptor(m.ProtoReflect().Descriptor())
+	if err != nil {
+		return fmt.Errorf("NormalizeDescriptor: %v", err)
+	}
+
+	// Instantiate a ManagedStream, which manages low level details like connection state and provides
+	// additional features like a future-like callback for appends, etc.  NewManagedStream can also create
+	// the stream on your behalf, but in this example we're being explicit about stream creation.
+	managedStream, err := client.NewManagedStream(ctx, managedwriter.WithStreamName(pendingStream.GetName()),
+		managedwriter.WithSchemaDescriptor(descriptorProto))
+	if err != nil {
+		return fmt.Errorf("NewManagedStream: %v", err)
+	}
+
+	// First, we'll append a single row.
+	rowBytes, err := generateExampleMessage()
+	if err != nil {
+		return fmt.Errorf("generateExampleMessage: %v", err)
+	}
+
+	// we'll keep track of the current offset in the stream with curOffset.
+	var curOffset int64
+
+	result, err := managedStream.AppendRows(ctx, [][]byte{rowBytes}, managedwriter.WithOffset(0))
+	if err != nil {
+		return fmt.Errorf("AppendRows first call error: %v", err)
+	}
+
+	// GetResult blocks until we receive a response from the API.
+	recvOffset, err := result.GetResult(ctx)
+	if err != nil {
+		return fmt.Errorf("AppendRows error appending at offset %d: %v", curOffset, err)
+	}
+	fmt.Fprintf(w, "Successfully appended data at offset %d.\n", recvOffset)
+
+	// Advance our current offset.
+	curOffset = curOffset + 1
+
+	// This time, we'll append three more rows in a single request.
+	result, err = managedStream.AppendRows(ctx, [][]byte{rowBytes, rowBytes, rowBytes}, managedwriter.WithOffset(curOffset))
+	if err != nil {
+		return fmt.Errorf("AppendRows second call error: %v", err)
+	}
+
+	// Block and wait for the result from the backend.
```

Comment:
Perhaps pretty clear from the term "block", but perhaps we call out that typically that this would be done in a separate thread/goroutine?

Reply:
Updated the comments around this to make it clearer.